### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by ZawwarAhmed**

### CODE REVIEW

File: Tests/EvenChallenge.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const evenChallenge = require("../Tasks/EvenChallenge");
+
+it("If the limit is set to 10, the program should output the following: 0 2 4 6 8 10", () => {
```
[the remark this answers is not in the document]
Comment (MuhammadAdam-VentureDive) [reply Tests/EvenChallenge.test.js:3]: test cases should contain corner cases as well like what happens if limit is -1 e.g..

**Review by MuhammadAdam-VentureDive — CHANGES_REQUESTED** [review]
I have reviewed only one task (even challenger) but the changes I suggested should be applied on all the task.
Try to name file or class with noun like evenChallenger.js should be named something like evenNumberGenerator.js
Please create FirstAssignment folder and move all the task into this folder.

File: Tasks/BMICalculator.js [hunk 2]
```diff
@@ -1,10 +1,9 @@
-const weight = 50;
-const height = 3;
-
-function BMI(weight, height) {
-  console.log((weight / (height * height)).toFixed(3));
+function getBMI(weight, height) {
+  if (height <= 0) return "height should be greater than zero."
```
Comment (MuhammadAdam-VentureDive) [inline Tasks/BMICalculator.js]: this function has side effects it is returning two different types it should return only value of BMI. If height and weight is not the expected input, handle through throw keyword

File: Tasks/Assignment 1/floydsTriangleGenerator.js [hunk 3]
```diff
@@ -1,16 +1,13 @@
-const n = 5;
-
 function floydsTriangle(n) {
+  if (n < 0) return "n should be greater than zero."
```
Comment (MuhammadAdam-VentureDive) [inline Tasks/Assignment 1/floydsTriangleGenerator.js:2]: please fix as I suggested in task 2

File: Tasks/Assignment 1/floydsTriangleGenerator.js [hunk 4]
```diff
@@ -1,16 +1,13 @@
-const n = 5;
-
 function floydsTriangle(n) {
```
Comment (MuhammadAdam-VentureDive) [inline Tasks/Assignment 1/floydsTriangleGenerator.js:1]: function name is not human readable

File: Tasks/Assignment 1/floydsTriangleGenerator.js [hunk 5]
```diff
@@ -1,16 +1,13 @@
...
+  if (n < 0) return "n should be greater than zero."
   let result = "";
   for (let i = 1; i <= n; i++) {
```
Comment (MuhammadAdam-VentureDive) [inline Tasks/Assignment 1/floydsTriangleGenerator.js:4]: this can be done with only one loop

File: Tasks/GradeAssignment.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+function getGrade(score) {
+  if (score > 100 || score < 0) return "Score should be between 0 and 100.";
```
Comment (MuhammadAdam-VentureDive) [inline Tasks/GradeAssignment.js]: fix this as I suggested in task 2

File: Tasks/GradeAssignment.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  else {
+    let grade =
```
Comment (MuhammadAdam-VentureDive) [inline Tasks/GradeAssignment.js]: make this logic configurable suppose that the value used in the logic is coming from database how would you implement in that case

Comment (ZawwarAhmed) [reply Tasks/GradeAssignment.js]: Please ellaborate this one please

Comment (MuhammadAdam-VentureDive) [reply Tasks/GradeAssignment.js]: there should not be if else statement. we should have to pass data  or globally create in this file. The data should be like [{ grade: A, from: 75, to: 80,}...]

Comment (ZawwarAhmed) [reply Tasks/GradeAssignment.js]: if there is no if else, then how would the function decide the grade?

Comment (MuhammadAdam-VentureDive) [reply Tasks/GradeAssignment.js]: this is the part you have to think how would the function decide the grade

Comment (MuhammadAdam-VentureDive) [reply Tasks/GradeAssignment.js]: If you are not able to find solution let me know I will give you a hint.

Comment (SyedZawwarAhmed) [reply Tasks/GradeAssignment.js]: I will find the solution.

File: Tasks/Assignment 1/gradeCalculator.js [hunk 8]
```diff
@@ -1,27 +1,53 @@
...
+    from: 0,
+    to: 50,
+  },
+];
 
-    return grade;
-  }
+function getGrade(score) {
+    let result = gradingCriteria.find(item => score >= item.from && score <= item.to);
+    if (result !== undefined)
```
Comment (MuhammadAdam-VentureDive) [inline Tasks/Assignment 1/gradeCalculator.js:46]: no need check with undefined you can do something like if(result) it will work

**Review by MuhammadAdam-VentureDive — CHANGES_REQUESTED** [review]

**Review by MuhammadAdam-VentureDive — APPROVED** [review]
LGTM !!!
